src/client.rs

```diff
@@ -175,32 +230,45 @@ impl Client {
     /// # Errors
     /// If an error occurs while fetching records, ```ClientError``` is returned
     pub fn get(&self) -> Result<Vec<KintoObject>, ClientError> {
-        let expected = get_latest_change_timestamp(
-            &self.server_url,
-            &self.bucket_name,
-            &self.collection_name,
-        )?;
 
-        let changeset = get_changeset(
-            &self.server_url,
-            &self.bucket_name,
-            &self.collection_name,
-            expected
-       )?;
-
-       debug!("changeset.metadata {}", serde_json::to_string_pretty(&changeset.metadata)?);
-
-       // verify the signature
-       let collection = Collection {
-           bid: self.bucket_name.to_owned(),
-           cid: self.collection_name.to_owned(),
-           metadata: changeset.metadata,
-           records: changeset.changes,
-           timestamp: changeset.timestamp
-        };
+        // If storage is empty for this bucket/collection, fetch from .../changeset, validate signature, store locally, and return records
+        // Otherwise, obtain current remote timestamp of collection from server (done in #26)
+        // If remote timestamp is different than local, fetch, validate signature, overwrite local existing data and return records
+        // If remote timestamp is same as local, validate signature and return records. If validating signature fails, fetch, validate, overwrite, and return records.
+
+        let key = format!("{}/{}", self.bucket_name, self.collection_name);
+        match self.cache.retrieve(&key) {
+            Ok(value) => {
+                // convert JSON object to LatestChangeResponse object
+                let cached_change_response: ChangesetResponse = serde_json::from_value(value)?;
```

Comment:
I'm confused. What do you store in the cache? The changeset response or the list of records?

Reply:
the changeset response